src/main/java/gov/nasa/jpl/mbee/mdk/emf/EMFExporter.java

```diff
@@ -307,6 +309,30 @@ private enum Processor {
                     return objectNode;
                 },
                 Type.POST
+        ),
+        DOCUMENTATION_POST(
+                (element, project, objectNode) -> {
+                    if (element != null) {
+                        Element docEl = Utils.getDocumentationElement(element);
+                        if (docEl != null) {
+                            ArrayNode ownedCommentIds = ((ArrayNode) objectNode.get(MDKConstants.OWNED_COMMENT_IDS_KEY));
+                            for (int i = 0; i < ownedCommentIds.size(); i++) {
+                                if (ownedCommentIds.get(i).asText().equals(getEID(docEl)))
+                                    ownedCommentIds.remove(i);
+                            }
+                            if (ownedCommentIds.size() == 0) {
+                                objectNode.remove(MDKConstants.OWNED_COMMENT_IDS_KEY);
```

Comment:
This is a bug fix If we don’t Magicdraw deletes the documentation.

Reply:
That is because we are overriding the default behavior of the export, and have not (yet) applied the inverse override in the import. During import, when pulling in MMS changes for `ownedCommentIds` we should inspect the Element's `ownedComments` and if the first one is the documentation element, then retain it (effectively adding it to the head of `ownedCommentIds`).

Reply:
Also I believe this proposed fix will only work in the case where there are no (real) `ownedComments`, i.e. only the documentation `Comment`. If there are any other user-created `ownedComments` it will still delete the documentation `Comment`.

Reply:
So far in testing it has worked as expected on both import and export (there is an import related fix too). If you are able to produce a specific issue we can try to resolve

Reply:
The issue also Ivan is that prior to MMS4 (which is when this was originally created). The "OWNED_COMMENT_IDS" field was not sent to MMS. So its not unreasonable to expect adding it now might cause issues